packages/gafl-webapp-service/src/routes/journey-definition.js

```diff
@@ -260,7 +260,13 @@ export default [
         page: CONTACT
       }
     },
-    backLink: s => (s.fromSummary === CONTACT_SUMMARY_SEEN ? CONTACT_SUMMARY.uri : LICENCE_FULFILMENT.uri)
+    backLink: s => {
+      if ((s.currentPage === LICENCE_FULFILMENT.page && s.fromSummary === CONTACT_SUMMARY_SEEN) || s.fromSummary !== CONTACT_SUMMARY_SEEN) {
+        return LICENCE_FULFILMENT.uri
+      } else {
+        return CONTACT_SUMMARY.uri
+      }
+    }
```

Comment:
Think this might get flagged by SonarCloud, but it should be
```
if ((s.currentPage === LICENCE_FULFILMENT.page && s.fromSummary === CONTACT_SUMMARY_SEEN) || s.fromSummary !== CONTACT_SUMMARY_SEEN) {
  return LICENCE_FULFILMENT.uri
}
return CONTACT_SUMMARY.uri
```
I also think that the criteria is a bit complex. A couple of well named variables might make it a bit clearer - `seenContactSummaryAndChoosingFulfilmentOption` and `notSeenContactSummary`.

Reply:
I can change it to this
```
const seenContactSummary = s.fromSummary === CONTACT_SUMMARY_SEEN
if ((s.currentPage === LICENCE_FULFILMENT.page && seenContactSummary ) || !seenContactSummary ) {
        return LICENCE_FULFILMENT.uri
}
return CONTACT_SUMMARY.uri
```